integration/emr/alluxio-emr.sh

```diff
@@ -280,179 +542,113 @@ USAGE_END
+      for i in "${input_args[@]}"; do
+        # forcibly escape the three characters " { }
+        esc_i=$(echo "${i}" | sed 's/"/\\"/g' | sed 's/{/\\{/g' | sed 's/}/\\}/g')
+        launch_args="${launch_args} ${esc_i}"
+      done
+    fi
+    bash -c "$launch_args" &
+    exit 0
   fi
+  echo "Executing asynchronous"
 
-  local -r workers=$(printf '%s\n' "${hostlist//$master/}")
-
-  doas alluxio "echo '${master}' > ${ALLUXIO_HOME}/conf/masters"
-  doas alluxio "echo '${workers}' > ${ALLUXIO_HOME}/conf/workers"
-
-  # set root ufs uri
-  if [[ "${root_ufs_uri}" = "LOCAL" ]]; then
-    root_ufs_uri="hdfs://${master}:8020"
-  fi
-
-  # Identify master
+  # collect instance information
+  local -r local_hostname=$(hostname -f)
   local -r is_master=$(jq '.isMaster' /mnt/var/lib/info/instance.json)
 
-  # Download files provided by "-f" to ${ALLUXIO_HOME}/conf
-  IFS=" " read -ra files_to_be_downloaded <<< "${files_list}"
-  if [ "${#files_to_be_downloaded[@]}" -gt "0" ]; then
-    local filename
-    for file in "${files_to_be_downloaded[@]}"; do
-      filename="$(basename "${file}")"
-      download_file "${file}"
-      sudo mv "${filename}" "${ALLUXIO_HOME}/conf/${filename}"
-    done
-    sudo chown -R alluxio:alluxio "${ALLUXIO_HOME}/conf"
-  fi
-  # Add newline to alluxio-site.properties in case the provided file doesn't end in newline
-  doas alluxio "echo >> ${ALLUXIO_SITE_PROPERTIES}"
-
-  if [[ "${delimited_properties}" ]]; then
-    # Inject user defined properties from args
-    IFS="${property_delimiter}" read -ra conf <<< "${delimited_properties}"
-    for property in "${conf[@]}"; do
-      local key=${property%%"="*}
-      local value=${property#*"="}
-      append_alluxio_property "${key}" "${value}"
-    done
+  # determine master hostname, different if on master vs worker
+  local master
+  if [[ "${is_master}" == "true" ]]; then
+    master="${local_hostname}"
+  else
+    master=$(jq '.masterHost' /mnt/var/lib/info/extraInstanceData.json | sed -e 's/^"//' -e 's/"$//' | nslookup | awk '/name/{print substr($NF,1,length($NF)-1)}')
   fi
 
-  local -r mem_size=$(get_default_mem_size)
-
-  # Query S3 for canonical ID of user and strip out the quotes
-  local -r canonical_id="$(aws s3api list-buckets --query "Owner.ID" | sed "s/\"//g")"
+  # wait until hadoop process is running
+  echo "Waiting for processes to start before starting script"
+  wait_for_hadoop "${is_master}" "${is_krb}"
 
+  echo "Starting Alluxio configuration"
+  set_custom_alluxio_properties "${delimited_properties}"
 
-  # Append default configs to site properties if the user hasn't set them already
+  # set auto generated properties
+  echo "Setting auto-generated properties in ${ALLUXIO_SITE_PROPERTIES}"
   doas alluxio "echo '# BEGIN AUTO-GENERATED PROPERTIES' >> ${ALLUXIO_SITE_PROPERTIES}"
   doas alluxio "echo '# Override these by specifying an alluxio-site.properties file, or delimited args within the EMR bootstrap' >> ${ALLUXIO_SITE_PROPERTIES}"
```

Comment:
This conflicts with our alluxio replacement strategy, consider removing it?

Reply:
hmm yeah probably should remove this echo statement